photon-ml/src/main/scala/com/linkedin/photon/ml/FeatureIndexingJob.scala

```diff
@@ -107,16 +132,24 @@ class FeatureIndexingJob(val sc: SparkContext,
     keyedFeaturesUnionedRDD.distinct().groupByKey(new HashPartitioner(partitionNum))
```

Comment:
Memory is a bit blurred here. I think previously I was worried that because I'm using `HashPartitioner` in group by key, and inside of HashPartitioner we also we do the hashing for a second time, so it's like `hash(hash(feature_str))`; because `hash(int) = int` I don't need to worry about in the reading stage, I cannot reproduce the hash code exactly; otherwise if `hash(int) != int` I might need to pass in some unique string key instead to make sure the hash code is reproducible and then string keys might be worse compared to int keys. (Yeah, the derivation is long here....)